File: services/core-api/app/schemas/notification.py
"""Notification-related Pydantic schemas."""

from datetime import datetime
from typing import Annotated
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.models.notification import NotificationType


class NotificationResponse(BaseModel):
    """Schema for notification response."""

    model_config = ConfigDict(from_attributes=True)

    id: UUID
    type: NotificationType
    payload: dict
    read: bool
    created_at: datetime


class NotificationListResponse(BaseModel):
    """Schema for notification list response."""

    items: list[NotificationResponse]
    unread_count: int
    total: int


class NotificationMarkReadRequest(BaseModel):
    """Schema for marking notifications as read."""

    notification_ids: Annotated[list[UUID], Field(min_length=1)]

File: services/frontend/src/types/notification.ts
/**
 * Notification-related type definitions
 */

export type NotificationType =
  | 'wishlist_shared'
  | 'item_marked'
  | 'item_unmarked'
  | 'item_resolved'
  | 'item_resolution_failed';

export interface Notification {
  id: string;
  type: NotificationType;
  payload: Record<string, string | null>;
  read: boolean;
  created_at: string;
}

export interface NotificationListResponse {
  items: Notification[];
  unread_count: number;
  total: number;
}
